src/ru/acomics/src/eu/kanade/tachiyomi/extension/ru/acomics/AComics.kt

```diff
@@ -22,177 +22,172 @@ class AComics : ParsedHttpSource() {
-        val manga = SManga.create()
-        manga.thumbnail_url = baseUrl + element.select("a > img").first()!!.attr("src")
-        element.select("div.title > a").first()!!.let {
-            manga.setUrlWithoutDomain(it.attr("href") + "/about")
-            manga.title = it.text()
-        }
-        return manga
-    }
-
-    override fun latestUpdatesFromElement(element: Element): SManga =
-        popularMangaFromElement(element)
-
-    override fun searchMangaFromElement(element: Element): SManga =
-        popularMangaFromElement(element)
-
-    override fun popularMangaNextPageSelector() = "span.button:not(:has(a)) + span.button > a"
-
-    override fun latestUpdatesNextPageSelector() = popularMangaNextPageSelector()
+    override fun searchMangaFromElement(element: Element) = popularMangaFromElement(element)
 
     override fun searchMangaNextPageSelector() = popularMangaNextPageSelector()
 
-    override fun mangaDetailsParse(document: Document): SManga {
-        val infoElement = document.select(".about-summary").first()!!
-        val manga = SManga.create()
-        manga.author = infoElement.select(".about-summary > p:contains(Автор)").text().split(":")[1]
-        manga.genre = infoElement.select("a.button").joinToString { it.text() }
-        manga.description = infoElement.ownText()
-        return manga
+    // =========================== Manga Details ============================
+    override fun mangaDetailsParse(document: Document) = SManga.create().apply {
+        val article = document.selectFirst("article.common-article")!!
+        with(article) {
+            title = selectFirst(".page-header-with-menu h1")!!.text()
+            genre = select("p.serial-about-badges a.category").joinToString { it.text() }
+            author = select("p.serial-about-authors a, p:contains(Автор оригинала)").joinToString { it.ownText() }
+            description = selectFirst("section.serial-about-text")?.text()
+        }
     }
 
+    // ============================== Chapters ==============================
     override fun chapterListParse(response: Response): List<SChapter> {
-        val res = mutableListOf<SChapter>()
-        val count = response.asJsoup()
-            .select(".about-summary > p:contains(Количество выпусков:)")
-            .text()
-            .split("Количество выпусков: ")[1].toInt()
-
-        for (index in count downTo 1) {
-            val chapter = SChapter.create()
-            chapter.chapter_number = index.toFloat()
-            chapter.name = index.toString()
-            val url = response.request.url.toString().split("/about")[0].split(baseUrl)[1]
-            chapter.url = "$url/$index"
-            res.add(chapter)
+        val doc = response.asJsoup()
+        val count = doc
+            .selectFirst("p:has(b:contains(Количество выпусков:))")!!
+            .ownText()
+            .toInt()
+
+        val comicPath = doc.location().substringAfter(baseUrl).substringBefore("/about")
```

Comment:
Avoid using `baseUrl` since the site might redirect to a new domain in the future.